website/src/docs/hotchocolate/defining-a-schema/extending-types.md

```diff
@@ -18,176 +16,340 @@ extend type Query {
+            .AddTypeExtension<BookExtensions>();
+    }
 }
 ```
 
-## Select types to extend
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
 
-Moreover, we can extend multiple types at once by extending upon base types or interfaces.
+**This is currently not working ([#3776](https://github.com/ChilliCream/hotchocolate/issues/3776))**
 
 ```csharp
-[ExtendObjectType(typeof(object))] // <-- we are now extending every type that inherits from object (essentially every type).
-public class SessionResolvers
+public class BookTypeExtensions : ObjectTypeExtension<Book>
+{
+    protected override void Configure(IObjectTypeDescriptor<Book> descriptor)
+    {
+        // this needs to match the name of the actual object type
+        descriptor.Name("Book");
+
+        descriptor
+            .Field(f => f.AuthorId)
+            .Type<AuthorType>()
+            .Name("author")
+            .Resolve(context =>
+            {
+                var parent = context.Parent<Book>();
+
+                // Omitted code for brevity
+            });
+    }
+}
+
+public class Startup
 {
-    public string SayHello() => "Hello";
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services
+            .AddGraphQLServer()
+            .AddTypeExtension<BookTypeExtensions>();
+    }
 }
 ```
 
-We can also extend multiple types at once with a type but dedicate specific resolvers to specific types.
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+Simply replace the field on the existing type.
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+## Extending base types
+
+We can also extend multiple types at once, but still dedicate specific resolvers to specific types.
 
 ```csharp
-[ExtendObjectType(typeof(object))] // <-- we are now extending every type that inherits from object (essentially every type)
-public class SessionResolvers
+// this extends every type that inherits from object (essentially every type)
+[ExtendObjectType(typeof(object))]
+public class ObjectExtensions
 {
-    public string Abc([Parent] Session session) => "abc"; // <-- we are only adding this field to the Session type
+    // this field is added to every object type
+    public string NewField()
+    {
+        // Omitted code for brevity
+    }
 
-    public string Def([Parent] Track track) => "def"; // <-- we are only adding this field to the Track type
+    // this field is only added to the Book type
+    public Author GetAuthor([Parent] Book book)
+    {
+        // Omitted code for brevity
+    }
+
+    // this field is only added to the Author type
+    public IEnumerable<Book> GetBooks([Parent] Author author)
+    {
+        // Omitted code for brevity
+    }
 }
 ```
 
-Instead of using `typeof(object)` as a selector for extending types you can also use interfaces or other base types.
-
-## Select types to extend with schema types
-
-We also can use schema types as a type selector.
+We can also modify all object types that are connected by an interface.
```

Comment:
```suggestion
We can also modify all object types that are connected by a base type, like an interface.

> Note: Below `IPost` is annotated with `[InterfaceType]` to include it in the GraphQL schema, but that isn't necessary for the type extension to work. You can use any base type, like `object`, an `abstract` base class, as an extension point without necessarily exposing the base type in your GraphQL schema.
```

Perhaps the note is unnecessary if the attribute is dropped from the sample, but just wanted to make it clearer that it's not req.